### ISSUE TITLE
`undocumented_unsafe_blocks` false positive on `const`s

### ISSUE THREAD

**Issue by jhpratt** [rust-lang/rust-clippy#11246]
### Summary

See repro, as it's self-explanatory.

### Lint Name

undocumented_unsafe_blocks

### Reproducer

```rust
#![deny(clippy::undocumented_unsafe_blocks)]

// Safety: foo
const _: () = unsafe {};
```

The lint fires, but it should not, as a safety comment is present on the preceding line. This is the case regardless of configuration.

### Version

```text
rustc 1.73.0-nightly (500647fd8 2023-07-27)
binary: rustc
commit-hash: 500647fd8138cc09e87edb08d62f81654fbf6ef8
commit-date: 2023-07-27
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```

**Comment by J-ZhengLi** [comment]
always wonder how this lint was able to check comments 🤔 , so I'm gonna try @rustbot claim

**Comment by J-ZhengLi** [comment]
as it turns out, there was a similar issue (#10832) which got fixed by adding a configuration to this lint.

Try adding a `accept-comment-above-statement = true` to your clippy.toml.

**Comment by jhpratt** [comment]
As I said in the original issue, this occurs regardless of configuration.

**Comment by J-ZhengLi** [comment]
> As I said in the original issue, this occurs regardless of configuration.

oops, I didn't read it through, my bad
